### PR TITLE
Create pipeline screen changes

### PR DESCRIPTION
**Pull request by praneshg239**
Screenshots:
### Create pipeline screen
image

### Create pipeline form modal
image

### Create pipeline form validation
image

### Create pipeline form details filled
image

### User navigated to pipeline edit screen once pipeline is created 
image

### CODE REVIEW

File: apps/gitness/src/pages/pipeline-create/PipelineCreateDialog/PipelineCreateForm.tsx [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+  const form = useZodForm({
+    schema: createPipelineSchema,
+    defaultValues: {
+      name: '',
+      branch: 'main',
+      yamlPath: ''
+    }
+  })
+
+  const { data, isLoading } = useListBranchesQuery({ repo_ref: 'MyNewProject/MyRepo/+', queryParams: {} })
```
Comment (abhinavrastogi-harness) [inline apps/gitness/src/pages/pipeline-create/PipelineCreateDialog/PipelineCreateForm.tsx:50]: we have the space and repo hooks now.. can we avoid hardcoded refs?

File: apps/gitness/src/pages/pipeline-create/PipelineCreateDialog/PipelineCreateForm.tsx [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+  const { watch, setValue, clearErrors, trigger } = form
```
Comment (abhinavrastogi-harness) [inline apps/gitness/src/pages/pipeline-create/PipelineCreateDialog/PipelineCreateForm.tsx:52]: can we destructure inline on line 41? makes code more readable imo

File: apps/gitness/src/pages/pipeline-create/PipelineCreateDialog/PipelineCreateForm.tsx [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+                <Input className="text-primary" {...field} />
+              </FormControl>
+              <FormMessage />
+            </FormItem>
+          )}
+        />
+
+        <FormField
+          control={form.control}
+          name="storeInGit"
```
Comment (abhinavrastogi-harness) [inline apps/gitness/src/pages/pipeline-create/PipelineCreateDialog/PipelineCreateForm.tsx:87]: is `storeInGit` required for gitness? i doubt it.. all pipelines will be in git by default

File: apps/gitness/src/pages/pipeline-create/PipelineCreateDialog/PipelineCreateForm.tsx [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+              <Checkbox checked={field.value} onCheckedChange={field.onChange} />
+            </FormControl>
+            <TooltipProvider>
+              <Tooltip delayDuration={200}>
+                <TooltipTrigger asChild>
+                  <FormLabel className="underline underline-offset-4 decoration-dotted" htmlFor="cloneRepo">
+                    Clone Git Repository
+                  </FormLabel>
+                </TooltipTrigger>
+                <TooltipContent className="bg-primary-foreground w-[350px] text-gray-100 border-gray-700">
```
Comment (abhinavrastogi-harness) [inline apps/gitness/src/pages/pipeline-create/PipelineCreateDialog/PipelineCreateForm.tsx:153]: avoid colors like `gray-100`.. use semantic tokens like `primary` or `secondary` instead

File: apps/gitness/src/pages/pipeline-create/PipelineCreateDialog/PipelineCreateForm.tsx [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+                  <p>
+                    If clone is enabled, Harness will clone your Git Repository automatically when the execution starts.
```
Comment (abhinavrastogi-harness) [inline apps/gitness/src/pages/pipeline-create/PipelineCreateDialog/PipelineCreateForm.tsx:155]: this is specific to Harness platform.. let's remove from gitness

File: apps/gitness/src/pages/pipeline-create/pipeline-create.tsx [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import { CreatePipelinePage } from '@harnessio/playground'
+import CreatePipelineDialog from './PipelineCreateDialog/PipelineCreateDialog'
+import { useState } from 'react'
+
+export function PipelineCreate() {
+  const [isSearchDialogOpen, setSearchDialogOpen] = useState(false)
```
Comment (abhinavrastogi-harness) [inline apps/gitness/src/pages/pipeline-create/pipeline-create.tsx:6]: naming... search?

File: apps/gitness/src/pages/pipeline-create/PipelineCreateDialog/PipelineCreateForm.tsx (hunk 2 above)
Comment (praneshg239) [reply apps/gitness/src/pages/pipeline-create/PipelineCreateDialog/PipelineCreateForm.tsx:52]: It is not props, it is form from `useZodForm`. As it required to be passed to `Form` component, we cannot destructure it.

File: apps/gitness/src/pages/pipeline-create/PipelineCreateDialog/PipelineCreateForm.tsx (hunk 3 above)
Comment (praneshg239) [reply apps/gitness/src/pages/pipeline-create/PipelineCreateDialog/PipelineCreateForm.tsx:87]: In the mock we had it. We can discuss and remove it

**Review by abhinavrastogi-harness — APPROVED** [review]

**Pull request merged**